frontend/src/hooks/payments/useBrowserStm/index.ts

```diff
@@ -0,0 +1,57 @@
+import { useLocalStorage } from '../../useLocalStorage'
+
+import {
+  addEntry,
+  deleteEntry,
+  deserialize,
+  getEntry,
+  processEviction,
+  serialize,
+  StmEntryDto,
+} from './utils'
+
+const PAYMENT_STM_KEY = 'PAYMENT_STM_KEY'
+/**
+ * In local storage, add a marker that a form submission and payment
+ * has been submitted and is ongoing. The marker must be cleared when
+ * the payment flow is complete. The marker also contains a eviction policy
+ * of maxmium 1 day, that will be assess whenever this hook is used.
+ *
+ * Returns an array of three variables:
+ *
+ * - `lastPaymentMemory` returns the previous value. It will be an empty string if
+ * there's no previous value in memory
+ *
+ * - `storePaymentMemory` expects the paymentId that will be stored. Replacing
+ * the previous value, if any.
+ *
+ * - `clearPaymentMemory` sets the memory to be an empty string
+ *
+ * @returns lastPaymentMemory
+ */
+export const useBrowserStm = (
+  formId: string,
+): [
+  lastPaymentMemory: StmEntryDto | undefined,
+  storePaymentMemory: (paymentId: string) => void,
+  clearPaymentMemory: () => void,
+] => {
+  const [paymentMemory, setPaymentMemory] = useLocalStorage(
+    PAYMENT_STM_KEY,
+    JSON.stringify({}),
+  )
+
+  const entryObj = deserialize(paymentMemory || '')
+  processEviction(entryObj)
```

Comment:
Bug here, you returned the returnObj from processEviction but did not assign it back to localstorage!

Hence, it won't get evicted (at least from local testing)

Reply:
Great catch.

After looking at it again, it's not idiomatic for the throttle function to return an object either. I've swapped the implementation in 4ea4c00a to apply a side effect instead.